src/app/icd/BUILD.gn

```diff
@@ -34,6 +34,8 @@ source_set("manager-srcs") {
     ":observer-srcs",
     "${chip_root}/src/credentials:credentials",
   ]
+
+  defines = [ "ICD_REPORT_ON_ENTER_ACTIVE_MODE=${chip_report_on_active_mode}" ]
```

Comment:
But how will this get propagated to ReportSchedulerImpl.cpp, which is where this thing is actually used?

Reply:
Since ReportScheduler inherits the ICDObserver, this define should be available for ReportSchedulerImpl no?

Reply:
Nope.  If you look at the actual compiler flags used with this PR, it's not in them.  See https://gn.googlesource.com/gn/+/main/docs/reference.md#var_defines for the rules about how "defines" are determined; if A depends on B, it will pick up defines from the public_configs and all_dependent_configs of B, but not defines set directly on B.

More generally, if `src/app/reporting/ReportSchedulerImpl.cpp` is the only thing using this define, shouldn't this define just be defined in the target that builds that file?

Reply:
Moved into src/app/BUILD.gn